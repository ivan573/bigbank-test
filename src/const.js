export const Route = {
  HOME: '/',
  GAME: '/game',
  SHOP: '/shop',
  REPUTATION: '/reputation'
}

export const ModalWindowType = {
  QUEST: 'quest',
  QUEST_RESULT: 'questResult',
  PURCHASE_ITEM: 'purchaseItem',
  PURCHASE_RESULT: 'purchaseResult',
  GAME_OVER: 'gameOver'
}
