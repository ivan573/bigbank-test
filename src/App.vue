<template>
  <div id="app" class="app">
    <h1 class="app__title">Dragons of Mugloar</h1>
    <nav v-if="gameId" class="app__navigation">
      <GameButton class="app__navigation-button" text="Main menu" :linkTo="homeRoute" primary />
      <GameButton class="app__navigation-button" text="Quests" :linkTo="questsRoute" primary />
      <GameButton class="app__navigation-button" text="Shop" :linkTo="shopRoute" primary />
      <GameButton class="app__navigation-button" text="Reputation" :linkTo="reputationRoute" primary />
    </nav>
    <GameData class="app__game-data" v-if="!isRouteHome" :data="data" />
    <router-view />
  </div>
</template>

<script>
import { mapState, mapGetters } from 'vuex'
import { Route } from '@/const'
import GameButton from '@/components/ui/GameButton.vue'
import GameData from '@/components/GameData.vue'

export default {
  name: 'App',
  components: {
    GameButton,
    GameData
  },
  data() {
    return {
      homeRoute: Route.HOME,
      questsRoute: Route.GAME,
      shopRoute: Route.SHOP,
      reputationRoute: Route.REPUTATION
    }
  },
  computed: {
    ...mapState('gameData', {
      gameId: state => state.gameId
    }),
    ...mapGetters('gameData', ['data']),
    isRouteHome() {
      return this.$route.path === Route.HOME
    }
  }
}
</script>

<style lang="scss" scoped>
.app {
  font-family: Arial, sans-serif;

  &__title {
    text-align: center;
  }

  &__navigation {
    margin-bottom: 16px;
    display: flex;
    gap: 8px;
    justify-content: center;
  }
}
</style>
