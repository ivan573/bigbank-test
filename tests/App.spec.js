import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import { Route } from '@/const'
import App from '@/App.vue'

describe('GameButton.vue', () => {
  const gameId = 'gameId'
  const data = {
    gold: 0,
    level: 0,
    lives: 3,
    score: 0,
    turn: 0
  }
  const gameData = {
    namespaced: true,
    state: {
      gameId: gameId
    },
    getters: {
      data() { return data }
    }
  }

  const notInitializedGameId = null
  const notInitializedData = {
    gold: 0,
    level: 0,
    lives: 0,
    score: 0,
    turn: 0
  }
  const notInitializedGameData = {
    namespaced: true,
    state: {
      gameId: notInitializedGameId
    },
    getters: {
      data() { return notInitializedData }
    }
  }

  it('matches snapshot', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)

    const store = new Vuex.Store({ modules: { gameData } })
    const router = new VueRouter()

    const wrapper = mount(App, { localVue, store, router })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders navigation if game is initialized', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)

    const store = new Vuex.Store({ modules: { gameData } })
    const router = new VueRouter()

    const wrapper = mount(App, { localVue, store, router })

    const navigationElement = wrapper.find('.app__navigation')

    expect(navigationElement.exists()).toBe(true)
  })

  it('does not render navigation if game is not initialized', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)

    const store = new Vuex.Store({ modules: { gameData: notInitializedGameData } })
    const router = new VueRouter()

    const wrapper = mount(App, { localVue, store, router })

    const navigationElement = wrapper.find('.app__navigation')

    expect(navigationElement.exists()).toBe(false)
  })

  it('renders game data if path not is root', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)

    const store = new Vuex.Store({ modules: { gameData } })
    const router = new VueRouter()

    const wrapper = mount(App, { localVue, store, router })

    const gameDataElement = wrapper.find('.app__game-data')

    expect(gameDataElement.exists()).toBe(false)
  })

  it('does not render game data if path is root', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)

    const store = new Vuex.Store({ modules: { gameData } })
    const router = new VueRouter()
    router.push(Route.GAME)

    const wrapper = mount(App, { localVue, store, router })

    const gameDataElement = wrapper.find('.app__game-data')

    expect(gameDataElement.exists()).toBe(true)
  })
})
