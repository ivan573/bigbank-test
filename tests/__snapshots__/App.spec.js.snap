// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameButton.vue matches snapshot 1`] = `
<div
  class="app"
  id="app"
>
  <h1
    class="app__title"
  >
    Dragons of Mugloar
  </h1>
   
  <nav
    class="app__navigation"
  >
    <button
      class="game-button app__navigation-button game-button_primary"
    >
      <span
        class="game-button__text"
      >
        Main menu
      </span>
    </button>
     
    <button
      class="game-button app__navigation-button game-button_primary"
    >
      <span
        class="game-button__text"
      >
        Quests
      </span>
    </button>
     
    <button
      class="game-button app__navigation-button game-button_primary"
    >
      <span
        class="game-button__text"
      >
        Shop
      </span>
    </button>
     
    <button
      class="game-button app__navigation-button game-button_primary"
    >
      <span
        class="game-button__text"
      >
        Reputation
      </span>
    </button>
  </nav>
   
  <!---->
   
  <!---->
</div>
`;
